tests/components/sonos/test_sensor.py

```diff
@@ -113,14 +118,45 @@ async def test_device_payload_without_battery_and_ignored_keys(
     assert ignored_payload not in caplog.text
 
 
-async def test_audio_input_sensor(hass, async_autosetup_sonos, soco):
+async def test_audio_input_sensor(
+    hass, async_autosetup_sonos, soco, device_properties_event
+):
     """Test audio input sensor."""
     entity_registry = ent_reg.async_get(hass)
 
+    speaker = list(hass.data[DATA_SONOS].discovered.values())[0]
```

Comment:
Can we patch something on the library instead so we don't need to access the speaker directly, which is an integration detail? Eg set the soco source and then receive a media event.